extensions/amp-base-carousel/1.0/arrow.js

```diff
@@ -34,16 +33,13 @@ export function Arrow({customArrow, by, advance, disabled}) {
     }
     advance(by);
   };
+  const classes = useStyles();
+  const classNames = `${classes.arrowPlacement} ${
+    by < 0 ? classes.arrowPrev : classes.arrowNext
+  } ${isDisabled ? classes.arrowDisabled : ''}`;
```

Comment:
Question: Do we prefer using `classes.arrowDisabled` for more precise specificity, or would we prefer defining a CSS rule for `${classes.arrowPlacement}[disabled]`? With the second approach we would also append `disabled={isDisabled}` as the prop.

Reply:
good question. not sure

Reply:
Here's my quick 2 cents on this. The most purist CSS way is `.x:disabled {}` - but that's not always applicable. The other options are `.x[disabled] {}` and `.x.x-disabled {}`. The important considerations here are:

1. The most important thing for us here is specificity. And all `.x:disabled`, `.x[disabled]` and `.x.x-disabled` all have exactly the same specificity (https://specificity.keegan.st/). So no clear winners there. However, the special case we can make `.x-disabled {}` that would include combined styles. This will have a lower specificity. But, I think this is a rare case where a lower specificity is actually harmful. I think the combined selector is a good thing.
2. `.x[disabled]` IMHO "feels" more semantic and self-documenting. But this is potentially less visible inside a component.
3. `.x.x-disabled` is faster, though in this case shouldn't be too big of a difference.
4. `.x.x-disabled` is slightly repetitive. We still need to add the `disabled` and `aria-disabled` attributes.

I think the bigger answer is that there's no a clear winner here. We can keep moving with a class and reassess/change later since these are fully compatible options due to the same specificity.